cmd/clair/main.go

```diff
@@ -180,10 +182,10 @@ func Boot(config *Config, slimMode bool) {
 	serv := server.New(fmt.Sprintf(":%d", config.API.HTTPSPort), db)
 	go api.RunClairify(serv)
 
-	grpcAPI := grpc.NewAPI(grpc.Config{
-		Port:         config.API.GRPCPort,
-		CustomRoutes: debugRoutes,
-	})
+	grpcAPI := grpc.NewAPI(
+		grpc.WithTLSEndpoint(config.API.GRPCPort),
+		grpc.WithDefaultInterceptors(),
+		grpc.WithCustomRoutes(debugRoutes))
```

Comment:
`grpc.NewAPI` is being called from both Node Inventory Mode and Scanner. However, they need different configurations of the GRPC server. 

1. Scanner requires TLS, Node inventory doesn't
2. Scanner requires a peer certificate interceptors, Node inventory doesn't

I thought it would be more idiomatic to use name opts function (e.g. `WithTLSEndpoint`) rather than passing multiple parameters or a struct with many properties.

Since I changed the `NewAPI` signature and pulled some dependencies outside (e.g. interceptors), I had to change how the grpc server is started on scanner path as well.